app/Filament/Resources/WaitingListResource/RelationManagers/AccountsRelationManager.php

```diff
@@ -117,11 +117,43 @@ public function table(Table $table): Table
                     ->visible(fn ($record) => $this->can('updateAccounts', $record->waitingList)),
 
                 Tables\Actions\ViewAction::make(),
-                Tables\Actions\DetachAction::make()
+                Tables\Actions\DetachAction::make('detachWithReason')
                     ->label('Remove')
-                    ->using(fn ($record, $livewire) => $livewire->ownerRecord->removeFromWaitingList($record->account))
-                    ->successNotificationTitle('User removed from waiting list'),
-            ])->defaultSort('created_at', 'asc')->persistSearchInSession()->defaultPaginationPageOption(25);
+                    ->form([
+                        Forms\Components\Select::make('reason')
+                            ->label('Reason for removal')
+                            ->options([
+                                'awarded_training_place' => 'Member awarded training place',
+                                'user_request' => 'Member requested removal',
+                                'other' => 'Other (specify below)',
+                            ])
+                            ->required()
+                            ->reactive(),
+
+                        Forms\Components\Textarea::make('custom_reason')
+                            ->label('Custom reason')
+                            ->rows(3)
+                            ->required()
+                            ->visible(fn (callable $get) => $get('reason') === 'other'),
+                    ])
+                    ->action(function (WaitingListAccount $record, array $data, $livewire) {
+                        $removal_reason = $data['reason'] === 'other' ? $data['custom_reason'] : $data['reason']; // if its other set the $removal_reason as the custom value (instead of 'other (specify below)')
```

Comment:
This perhaps highlights the need to distinguish the reason vs the comment. 

removed_by
removal_type - this effectively becomes an enum
removal_comment

You may not need the comment to be attached to the record in the same table, it could instead be a note on the account which I believe should be polymorphic.